semgrep_output_v1.atd

```diff
@@ -1680,6 +1686,39 @@ type ci_scan_complete_stats = {
     <python repr="dict">
     <ts repr="map">
     option;
+
+  (* since 1.98.0 *)
+  (* In collaboration with the Data Science team, it was suggested
+   * that we start to group stats by product for organizational purposes.
+   *)
+  ?supply_chain_stats: supply_chain_stats option;
+}
+
+(* Valid schema versions. Update when making breaking changes. *)
+type stats_schema_version <ocaml attr="deriving show"> <python decorator="dataclass(frozen=True)"> = [
+  | SSV20241112 (* initial version *)
+] 
+
+type resolution_method <ocaml attr="deriving show"> <python decorator="dataclass(frozen=True, order=True)"> = [
```

Comment:
@bkettle we have two columns, `lockfile_flag` and `manifest_flag`, in the [proposed SSC fact table](https://docs.google.com/spreadsheets/d/14jNMAvE2YRrnpmRPzodIIjohM9s9iLUO/edit?gid=1327685153#gid=1327685153). 

I considered adding `ManifestParsing` to specify whether we parsed a lockfile or a manifest file, but that may be redundant.

For tracking, Khoa wants to distinguish `dependency_source_files` as either lockfiles or manifest files. Currently, we’re using `LockfileParsing` or `DynamicResolution`. It seems we can map `LockfileParsing` to `lockfile_flag` and `DynamicResolution` to `manifest_flag` (since lockfileless setups depend on some kind of manifest file, correct?).

Reply:
Lockfileless setups might not depend on a manifest file in the future---e.g. for Go, there is only a `go.mod` file that serves both as a manifest and a lockfile. We will probably categorize this as a lockfile, but PTT would require it be resolved with dynamic resolution. Might not be worth trying to build for that now, though.

What's the purpose of the `lockfile_flag` vs `manifest_flag`?

I wonder if we should consider using the new `dependency_source` type to be introduced in https://github.com/semgrep/semgrep-interfaces/pull/312 in the stats JSON?

Reply:
> Lockfileless setups might not depend on a manifest file in the future

Makes sense, thanks for the context

> What's the purpose of the lockfile_flag vs manifest_flag?

I believe it's just used for tracking purposes -- just to have more metadata available. 

> I wonder if we should consider using the new dependency_source type to be introduced in https://github.com/semgrep/semgrep-interfaces/pull/312 in the stats JSON?

Ohhh. that would be helpful actually.

Reply:
OK, there is a bit more work to do there because that PR needs to be merged in sync with the `semgrep-pro` PR, and there are a couple things left to tweak in the `semgrep-pro` PR, but I am hoping to make it happen tomorrow or the day after.

Reply:
perfect, I have plenty to do in the meantime!

Reply:
@bkettle using new dependency type here [fb81785](https://github.com/semgrep/semgrep-interfaces/pull/313/commits/fb81785f2e43ad0c3166602090a43e0e8c044278)